doc/library/scan.txt

```diff
@@ -454,6 +457,78 @@ As a rule, scan always expects the condition to be the last thing returned
 by the inner function, otherwise an error will be raised.
 
 
+Optimizing Scan's performance
+-----------------------------
+
+This section covers some ways to improve performance of a Theano function
+using Scan.
+
+
+Minimizing Scan usage
+^^^^^^^^^^^^^^^^^^^^^
+
+Scan makes it possible to define simple and compact graphs that can do the
+same work as much larger and more complicated graphs. However, it comes with
+a significant overhead. As such, when performance is the objective, a good
+rule of thumb is to perform as much of the computation as possible outside of
+Scan. This may have the effect of increasing memory usage but can also
+reduce the overhead introduces by using Scan.
+
+
+Passing shared variables as non-sequences
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+It is possible, inside of Scan, to use shared variables previously defined
+outside of the Scan without explicitly passing them as inputs to the Scan.
+However it is often more efficient to explicitly pass them as non-sequence
+inputs instead. Section :ref:`lib_scan_shared_variables` provides an
```

Comment:
Actually, passing the shared variables themselves as non-sequences will not help performance.
What can help is if a variable depending only on shared variables (and constants) is explicitly passed as a non-sequence, instead of its shared ancestors being (implicitly) the non-sequences.